Dockerfile

```diff
@@ -39,19 +39,49 @@ RUN curl --silent https://packagecloud.io/install/repositories/cs50/repo/script.
 RUN curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash && \
     apt-get install -y git-lfs
 
-# Install Python packages
-RUN apt-get update && apt-get install -y python3-pip && \
-    pip3 install \
+# Install Python 3.7
+ENV PYENV_ROOT /opt/pyenv
+RUN apt-get update && \
+    apt-get install -y \
+        build-essential \
+        curl \
+        libbz2-dev \
+        libffi-dev \
+        libncurses5-dev \
+        libncursesw5-dev \
+        libreadline-dev \
+        libsqlite3-dev \
+        libssl-dev \
+        llvm \
+        make \
+        tk-dev \
+        wget \
+        xz-utils \
+        zlib1g-dev && \
+    wget -P /tmp https://github.com/pyenv/pyenv/archive/master.zip && \
+    unzip -d /tmp /tmp/master.zip && \
+    rm -f /tmp/master.zip && \
+    mv /tmp/pyenv-master "$PYENV_ROOT" && \
```

Comment:
Any reason not to `git clone` into `PYENV_ROOT` directly?

Reply:
Didn't want the `.git` to be there (though we could just `rm`). And just tend to do this out of habit, ensures that the `mv` is atomic, so the dir's either there or not, not partially populated. Doesn't matter in a `Dockerfile` so much but in a script, feels best to move things into place only when ready.